src/zappy_ai_src/ai_strategy_management.py

```diff
@@ -0,0 +1,430 @@
+            When availableSlots == 0 it means that the team is full and cannot accept more connection.
+        """
+        self.__availableSlots = availableSlots
+
+        """Every team got a name which is used to separate different player in different group"""
+        self.__teamName = teamName
+
+        """This is the wrapper of the library used to communicate with the server"""
+        self.__lib = ServerWrapper("./libzappy_ai_api.so")
+
+        """This boolean is used to know the current state of the AI
+            True if the AI is looping and making some actions
+            False if the player died or client disconnected
+        """
+        self.__isRunning = False
+
+        """This is a temporary inventory used in strategies to plan actions of player in different way
+            As an example if a player runs out of food, the survival action will take priority
+        """
+        self.__inventory = Inventory("food 120, linemate 0, deraumere 0, sibur 0, mendiane 0, phiras 0, thystame 0")
+
+        """This is an ephemeral representation of the player's view of the map.
+            Of course, the player's vision depends on his current level
+            Initialized with blank string by default.
+            This private member can be useful to develop and use relevant action from the AI
+        """
+        self.__visionOfTheMap = Map("")
+
+        """This private member represents the index of the current target tile of the AI.
+            It can be useful to use relevant action from the AI in component research as example.
+            Set as int: -1 by default.
+        """
+        self.__targetTileIndex = -1
+
+        """This private member represents the currently most needed component by the AI.
+            It can be useful to use relevant action from the AI in component research as example.
+            Set as str: 'food' by default
+        """
+        self.__targetComponent = "food"
+
+        """This private member represents the current level of the player
+            It impacts on his vision of the map and his requirements of elevation
+            Set as int: 1 by default and can reach level 8 maximum
+        """
+        self.__playerCurrentLevel = 1
+
+        """This is mandatory call by its creator"""
+        self.__lib.getNecessaryFunctions()
+
+    def __del__(self):
+        """Default Destructor of the Core class"""
+        self.running = False
+
+    """----------------------------------------Getter and Setter for AI class----------------------------------------"""
+
+    def __setAvailableSlots(self, availableSlots: int):
+        self.__availableSlots = availableSlots
+
+    def __setIsRunning(self, isRunning: bool):
+        self.__isRunning = isRunning
+
+    """
+        [ASKING], should i delete the old inventory and then re use the constructor the refill the inventory,
+        or should we turn the method fill inventory as public
```

Comment:
I think we should turn the object fill inventory as public, do it in this PR

Reply:
yes good idea